databases/week2/createDatabaseBusiness.sql

```diff
@@ -0,0 +1,145 @@
+CREATE DATABASE business;
+Use business;
+CREATE TABLE `staff` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `duty` varchar(255) NOT NULL,
+  `status_id` int(10) unsigned NOT NULL,
+  CONSTRAINT `fk_status` FOREIGN KEY (`status_id`) REFERENCES `status` (`id`) ON DELETE CASCADE ON UPDATE CASCADE PRIMARY KEY (`id`)
+);
+CREATE TABLE `jobTitle` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `jobTitle` varchar(255) NOT NULL,
+  PRIMARY KEY (`id`)
+);
+CREATE TABLE `status` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` varchar(255) NOT NULL
+);
+CREATE TABLE `office` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `place` varchar(255) NOT NULL,
+  PRIMARY KEY (`id`)
+);
+CREATE TABLE `staff_office` (
+  `staff_id` int(10) unsigned NOT NULL,
+  `office_id` int(10) unsigned NOT NULL,
+  PRIMARY KEY(`staff_id`, `office_id`),
+  CONSTRAINT `fk_staff_office_staff` FOREIGN KEY (`staff_id`) REFERENCES `staff` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_staff_office_office` FOREIGN KEY (`office_id`) REFERENCES `office` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
The `ON DELETE CASCADE` functionality will delete all the referenced entries when something is deleted. 
e.g. If you delete an _office_, all of the connected _staff_ will also be deleted and the same way around. 

This can be very useful in certain cases but sometimes it can also have a negative impact

Reply:
okay...